File: core/src/main/java/javabot/commands/InfoApi.java
package javabot.commands;

import com.antwerkz.sofia.Sofia;
import javabot.Message;
import javabot.dao.ApiDao;
import javabot.javadoc.JavadocApi;

import javax.inject.Inject;

public class InfoApi extends AdminCommand {
    @Inject
    private ApiDao dao;

    @Param
    String name;

    @Override
    public void execute(final Message event) {
        final JavadocApi api = dao.find(name);
        if (api != null) {
            getBot().postMessageToChannel(event, Sofia.apiLocation(api.getName(), api.getBaseUrl()));
        } else {
            getBot().postMessageToChannel(event, Sofia.unknownApi(name, event.getUser().getNick()));
        }
    }
}

File: src/main/kotlin/javabot/commands/InfoApi.kt
package javabot.commands

import com.antwerkz.sofia.Sofia
import com.beust.jcommander.Parameter
import javabot.Javabot
import javabot.Message
import javabot.dao.AdminDao
import javabot.dao.ApiDao
import javax.inject.Inject

class InfoApi @Inject constructor(bot: Javabot, adminDao: AdminDao, var apiDao: ApiDao) : AdminCommand(bot, adminDao) {

    @Parameter(required = true)
    lateinit var apiName: String

    override fun execute(event: Message): List<Message> {
        val responses = arrayListOf<Message>()
        val api = apiDao.find(apiName)
        if (api != null) {
            responses.add(Message(event, Sofia.apiLocation(api.name, api.version)))
        } else {
            responses.add(Message(event, Sofia.unknownApi(apiName, event.user.nick)))
        }
        return responses
    }
}